src/tbp/monty/frameworks/models/monty_base.py

```diff
@@ -437,11 +437,11 @@ def get_agent_state(self):
         Returns:
             State of the agent.
         """
-        return self.motor_system._policy.get_agent_state()
+        return self.motor_system._policy.get_agent_state(self.motor_system._state)
```

Comment:
Why do we need to call `get_agent_state` on the policy by passing it the state? Wouldn't it make much more sense to do `return self.motor_system._state` or `return self.motor_system.get_agent_state()`? If it is just because of the `agent_id` property of the policy, couldn't that be extracted in a `get_agent_state` function of the motor system? like `return self._state[self.policy.agent_id]`

Reply:
okay that makes a lot of sense. Feel free to keep track of this however works best for you. If you add a TODO it would make it more obvious to me that you are already planning on changing this and I won't feel compelled commenting on it. But if you already have another system of tracking this I don't want to create any overhead for you.